### DISCUSSION TITLE
Duplicate slug incrementing

### DISCUSSION THREAD

**Discussion by 37b** [radiac/django-tagulous#146] [Q&A] [answered]
I am using tagulous indirectly as a user of [Defect Dojo](https://github.com/DefectDojo/django-DefectDojo) and I am having a problem.

I have multiple tags that all begin with the same _n_ characters which get truncated to something like this for the slug

`urnasset1234content-definition-testcycle-managemen`
`urnasset1234content-definition-testcycle-mana_1`
`urnasset1234content-definition-testcycle-mana_2`
`urnasset1234content-definition-testcycle-mana_3`
`urnasset1234content-definition-testcycle-mana_4`
`urnasset1234content-definition-testcycle-mana_5`
`urnasset1234content-definition-testcycle-mana_6`
`urnasset1514content-definition-testcycle-mana_7`
`urnasset1514content-definition-testcycle-mana_8`
`urnasset1514content-definition-testcycle-mana_9`
`urnasset1514content-definition-testcycle-mana_10`

However, once I hit `_10` the app no longer increments and instead throws an exception

`django.db.utils.IntegrityError: (1062, "Duplicate entry 'urnasset1514content-definition-lifecycle-mana_10' for key 'dojo_tagulous_engagement_tags_slug_53c6df0a_uniq'`

Is there a way to increase this number of increments? If not, can I at least lengthen the slug? I trying doing this at the DB level but it did not bubble up to the app.

**Comment by radiac** [comment, marked as answer]
Looks like this was a bug in the unique slug generator, should be fixed in #152.

The system should now be good up to 10,000 unique slugs now - the algorithm uses a hardcoded truncate length of 5, one for the underscore and 4 for the numbers.

The slugs are intended for human-readable (and SEO-friendly) URLs; if anyone approaches 10,000 collisions then arguably it's not going to be a useful slug any more. I think the best thing at that point would be to look at adding support for custom slugging algorithms, as I can't imagine many use cases at that point so it's probably safest to say it's out of scope.
